realm/kotlin-extensions/src/androidTest/kotlin/io/realm/CoroutinesTests.kt

```diff
@@ -924,6 +928,160 @@ class CoroutinesTests {
         TestHelper.awaitOrFail(countDownLatch)
     }
 
+    @Test
+    fun realmList_flow_parentDeletionComplete() = looperThread.runBlocking {
+        val realm = Realm.getInstance(configuration)
+        realm.beginTransaction()
+        val parent = realm.createObject<AllTypes>()
+        val list = parent.columnRealmList
+        list.add(Dog("dog"))
+        realm.commitTransaction()
+
+        val collectingStarted = AtomicBoolean(false)
+        val context = looperThread.asDispatcher()
+        val scope = CoroutineScope(context)
+        scope.launch {
+            // We should only emit valid lists. If the parent of the list is invalidated
+            // it should close the stream gracefully resulting in onComplete being called.
+            list.toFlow()
+                    .onEach { assertTrue(it.isValid) }
+                    .onCompletion {
+                        realm.close()
+                        looperThread.testComplete()
+                    }.collect {
+                        collectingStarted.set(true)
```

Comment:
Maybe not required for now, but I guess you could communicate through a channel now that we are in a coroutine context. That would make it cleaner when waiting for the result. Good old CSP.